### ISSUE TITLE
Unhandled rejection TypeError: next is not a function

### ISSUE THREAD

**Issue by shashikiran797** [joegoldbeck/mongoose-encryption#75]
I recently updated my mongoose-encryption version from 1.4 to 1.5 and did not make any code changes.

And I get the following error when I try to query the encrypted fields

```
0|qa2      | Unhandled rejection TypeError: next is not a function
0|qa2      |     at model.<anonymous> (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongoose-encryption/lib/plugins/mongoose-encryption.js:267:24)
0|qa2      |     at Kareem.execPreSync (/home/ubuntu/Work/alliswell_nodeapp/node_modules/kareem/index.js:115:16)
0|qa2      |     at model.syncWrapper [as $__init] (/home/ubuntu/Work/alliswell_nodeapp/node_modules/kareem/index.js:232:12)
0|qa2      |     at model.Document.init (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongoose/lib/document.js:393:8)
0|qa2      |     at completeMany (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongoose/lib/query.js:1463:14)
0|qa2      |     at cb (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongoose/lib/query.js:1322:11)
0|qa2      |     at result (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/utils.js:414:17)
0|qa2      |     at executeCallback (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/utils.js:406:9)
0|qa2      |     at handleCallback (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/utils.js:128:55)
0|qa2      |     at self.close (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:904:60)
0|qa2      |     at handleCallback (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/utils.js:128:55)
0|qa2      |     at completeClose (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:1043:14)
0|qa2      |     at _endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:1053:35)
0|qa2      |     at Cursor._endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:190:5)
0|qa2      |     at Cursor._endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:217:59)
0|qa2      |     at Cursor.close (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:1053:17)
0|qa2      |     at /home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:904:21
0|qa2      |     at handleCallback (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:199:5)
0|qa2      |     at _setCursorNotifiedImpl (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:556:38)
0|qa2      |     at self._endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:564:46)
0|qa2      |     at ClientSession.endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/sessions.js:72:41)
0|qa2      |     at Cursor._endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:185:13)
0|qa2      |     at Cursor._endSession (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb/lib/cursor.js:217:59)
0|qa2      |     at _setCursorNotifiedImpl (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:564:17)
0|qa2      |     at setCursorNotified (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:556:3)
0|qa2      |     at setCursorDeadAndNotified (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:549:3)
0|qa2      |     at nextFunction (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:769:5)
0|qa2      |     at Cursor.next (/home/ubuntu/Work/alliswell_nodeapp/node_modules/mongodb-core/lib/cursor.js:810:3)
```

**Comment by virginielgb** [comment]
Hi,
Downgrading mongoose to 4.x should do the trick. Devs are already working on it, see https://github.com/joegoldbeck/mongoose-encryption/issues/70

**Comment by shashikiran797** [comment]
I will give it a try, thanks for the reply @virginielgb

**Comment by helgetan** [comment]
any updates here?

**Comment by shashikiran797** [comment]
@helgetan 
Downgrading the mongoose version to 4.13.12 worked for me

**Comment by ChrGriffin** [comment]
Using mongoose ^5.1.4 gave me this error. Downgrading to 4.13.14 solved the issue.

**Comment by zacharybrady** [comment]
Same. Can't downgrade. Any update coming for this module?

**Comment by colinhemphill** [comment]
We have lots of ongoing discussion on Mongoose v5 support.

https://github.com/joegoldbeck/mongoose-encryption/issues/70
https://github.com/joegoldbeck/mongoose-encryption/pull/71

**Comment by joegoldbeck** [comment]
Fixed as of 2.0.0 https://github.com/joegoldbeck/mongoose-encryption/tree/2.0.0
